src/System.Threading.Overlapped/tests/OverlappedTests.cs

```diff
@@ -0,0 +1,166 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xunit;
+
+public static partial class OverlappedTests
+{
+    [Fact]
+    public static void PropertyTest1()
+    {
+        IAsyncResult asyncResult = new Task(() => Console.WriteLine("this is a dummy task"));
+        var obj = new Overlapped();
+
+        Assert.Null(obj.AsyncResult);
+        obj.AsyncResult = asyncResult;
+        Assert.Same(obj.AsyncResult, asyncResult);
+
+#pragma warning disable 618
+        Assert.Equal(obj.EventHandle, 0);
+        obj.EventHandle = 3;
+        Assert.Equal(obj.EventHandle, 3);
+#pragma warning restore 618
+
+        var _handle = new ManualResetEvent(false).SafeWaitHandle;
+        Assert.NotSame(obj.EventHandleIntPtr, IntPtr.Zero);
+        obj.EventHandleIntPtr = _handle.DangerousGetHandle();
+        Assert.Equal(obj.EventHandleIntPtr, _handle.DangerousGetHandle());
+
+        Assert.Equal(obj.OffsetHigh, 0);
+        obj.OffsetHigh = 3;
+        Assert.Equal(obj.OffsetHigh, 3);
+
+        Assert.Equal(obj.OffsetLow, 0);
+        obj.OffsetLow = 1;
+        Assert.Equal(obj.OffsetLow, 1);
+    }
+
+    [Fact]
+    public static void PropertyTest2()
+    {
+        IAsyncResult asyncResult = new Task(() => Console.WriteLine("this is a dummy task"));
+        var _event = new ManualResetEvent(false);
+        var _handle = _event.SafeWaitHandle;
+
+#pragma warning disable 618
+        var obj = new Overlapped(1, 3, _event.Handle.ToInt32(), asyncResult);
+        Assert.Equal(obj.EventHandle, _event.Handle.ToInt32());
+#pragma warning restore 618
+
+        Assert.Same(obj.AsyncResult, asyncResult);
+        Assert.Equal(obj.EventHandleIntPtr, _handle.DangerousGetHandle());
+        Assert.Equal(obj.OffsetHigh, 3);
+        Assert.Equal(obj.OffsetLow, 1);
+    }
+
+    [Fact]
+    public static void PropertyTest3()
+    {
+        IAsyncResult asyncResult = new Task(() => Console.WriteLine("this is a dummy task"));
+        var _event = new ManualResetEvent(false);
+        var _handle = _event.SafeWaitHandle;
+        var obj = new Overlapped(1, 3, _handle.DangerousGetHandle(), asyncResult);
+        Assert.Same(obj.AsyncResult, asyncResult);
+
+#pragma warning disable 618
+        Assert.Equal(obj.EventHandle, _event.Handle.ToInt32());
+#pragma warning restore 618
+
+        Assert.Equal(obj.EventHandleIntPtr, _handle.DangerousGetHandle());
+        Assert.Equal(obj.OffsetHigh, 3);
+        Assert.Equal(obj.OffsetLow, 1);
+    }
+
+    [Fact]
+    public unsafe static void PackNegTest()
+    {
+        var helper = new AsyncHelper();
+        IOCompletionCallback callback = MyCallback(helper);
+
+        NativeOverlapped* nativeOverlapped;
+        Overlapped ov = new Overlapped();
+        nativeOverlapped = ov.Pack(new IOCompletionCallback(callback), null);
```

Comment:
I think you meant the call to free. unpack only changes the representation.
correct ?